pkg/apicast/apicast_options.go

```diff
@@ -46,3 +48,16 @@ func (a *APIcastOptions) Validate() error {
 	validate := validator.New()
 	return validate.Struct(a)
 }
+
+func DefaultResourceRequirements() v1.ResourceRequirements {
```

Comment:
This seems it is not being use anywhere. I understand we should use this and set it as the apicast's resourcerequirements when the user has not specified resources in the CR